src/components/site-content-tabs.tsx

```diff
@@ -27,22 +28,24 @@ export function SiteContentTabs() {
 	return (
 		<div className="flex flex-col w-full h-full app-no-drag-region pt-8 overflow-y-auto">
 			<Header />
-			<TabPanel
-				className="mt-6 h-full flex flex-col overflow-hidden"
-				tabs={ tabs }
-				orientation="horizontal"
-			>
-				{ ( { name } ) => (
-					<div className="h-full">
-						{ name === 'overview' && <ContentTabOverview selectedSite={ selectedSite } /> }
-						{ name === 'share' && <ContentTabSnapshots selectedSite={ selectedSite } /> }
-						{ name === 'settings' && <ContentTabSettings selectedSite={ selectedSite } /> }
-						{ assistantEnabled && name === 'assistant' && (
-							<ContentTabAssistant selectedSite={ selectedSite } />
-						) }
-					</div>
-				) }
-			</TabPanel>
+			<WelcomeMessagesProvider>
+				<TabPanel
+					className="mt-6 h-full flex flex-col overflow-hidden"
+					tabs={ tabs }
+					orientation="horizontal"
+				>
+					{ ( { name } ) => (
+						<div className="h-full">
+							{ name === 'overview' && <ContentTabOverview selectedSite={ selectedSite } /> }
+							{ name === 'share' && <ContentTabSnapshots selectedSite={ selectedSite } /> }
+							{ name === 'settings' && <ContentTabSettings selectedSite={ selectedSite } /> }
+							{ assistantEnabled && name === 'assistant' && (
+								<ContentTabAssistant selectedSite={ selectedSite } />
+							) }
+						</div>
+					) }
+				</TabPanel>
+			</WelcomeMessagesProvider>
```

Comment:
Not a strong opinion, but what do you think about moving the provider to wrap only the ContentTabAssistant ? 

```diff
-		<WelcomeMessagesProvider>
			<TabPanel
				className="mt-6 h-full flex flex-col overflow-hidden"
				tabs={ tabs }
				orientation="horizontal"
			>
				{ ( { name } ) => (
					<div className="h-full">
						{ name === 'overview' && <ContentTabOverview selectedSite={ selectedSite } /> }
						{ name === 'share' && <ContentTabSnapshots selectedSite={ selectedSite } /> }
						{ name === 'settings' && <ContentTabSettings selectedSite={ selectedSite } /> }
+						<WelcomeMessagesProvider>
							{ assistantEnabled && name === 'assistant' && (
								<ContentTabAssistant selectedSite={ selectedSite } />
							) }
+						</WelcomeMessagesProvider>
					</div>
				) }
			</TabPanel>
-		</WelcomeMessagesProvider>
```

Reply:
I originally placed this component at this location but realized it was being rendered on every navigation to the Assistant tab. This is the main reason why I moved it to an upper rendering level.